DealerInspireLaravelCodingStandard/Sniffs/Providers/DeferredProvidersSniff.php

```diff
@@ -116,21 +121,33 @@ public function process(File $phpcsFile, $stackPtr): void
      */
     protected function isDeferred(int $index, array $tokens): ?bool
     {
+        if ($tokens[$index]['code'] === T_IMPLEMENTS) {
+            $this->checkingForDeferrableProvider = true;
+            return null;
+        }
+
         if ($tokens[$index]['code'] === T_VARIABLE && $tokens[$index]['content'] === '$defer') {
             $this->checkingForDeferredValue = true;
+            return null;
         }
 
-        if ($this->checkingForDeferredValue === false) {
-            return null;
+        if ($this->checkingForDeferrableProvider) {
+            // The first string after 'implements' will be the value
+            if ($tokens[$index]['code'] === T_STRING) {
+                $this->checkingForDeferrableProvider = false;
```

Comment:
If we search until we find a T_OPEN_CURLY_BRACKET, we could also allow this to detect when DeferrableProvider is the 2nd+ interface being implemented.

Something akin to 
```
if ($this->checkingForDeferrableProvider) {
    if ($tokens[$index]['code'] === T_OPEN_CURLY_BRACKET) {
        // We've searched all the implemented interfaces
        $this->checkingForDeferrableProvider = false;
        return null;
    }
    if ($tokens[$index]['code'] === T_STRING && $tokens[$index]['content'] === 'DeferrableProvider') {
        $this->checkingForDeferrableProvider = false;
        return true;
    }
}
```

Reply:
That's really smart, I'll implement that here soon!